lua/entities/gmod_tardis/modules/sh_artronenergy.lua

```diff
@@ -0,0 +1,311 @@
+if SERVER then
+
+    local artron_values = {
+        cost_demat = -540,
+        cost_mat = -360,
+        cost_full = -1200,
+        cost_controls = {
+            ["cloak"] = -720,
+            ["physlock"] = -540,
+            ["float"] = -540,
+        },
+        cost_hads = -180,
+        cost_failed_demat = -80,
+
+        -- every 1 second:
+        spend_vortex_teleport = -32,
+        spend_flight = -45,
+        spend_flight_boost = -45,
+        spend_cloak = -36,
+        spend_cloak_handbrake = -12,
+
+        -- every 5 seconds:
+        increase_handbrake = 180 * 5,
+        increase_normal = 15 * 5,
+        increase_warning = 10 * 5,
+        increase_poweroff = 24 * 5,
+        increase_float = 8 * 5,
+        increase_float_handbrake = 40 * 5,
+
+        increase_engine_release = 720,
+    }
+
+    function ENT:SetArtron(value)
+        local maxArtron = TARDIS:GetSetting("artron_energy_max")
+        value = math.max(0, math.min(value, maxArtron))
+        self:SetData("artron-val", value, true)
+        if value == 0 then
+            self:CallHook("ArtronDepleted")
+        end
+        tardisdebug(value)
+    end
+
+    function ENT:AddArtron(value)
+        local currentArtron = self:GetData("artron-val", 0)
+        self:SetArtron(currentArtron + value)
+    end
+
+    ENT:AddHook("Initialize", "artron", function(self)
+        local max = TARDIS:GetSetting("artron_energy_max")
+        local start = TARDIS:GetSetting("artron_energy_start") -- 0, 0.25, 0.5, 0.75 or 1
+        self:SetArtron(max * start)
+    end)
+
+
+    --
+    -- Bad artron consequences
+    --
+
+    ENT:AddHook("ArtronDepleted", "teleport_and_poweroff", function(self)
+        if not TARDIS:GetSetting("artron_energy") then return end
+
+        if self:GetData("teleport") or self:GetData("vortex") then
+            self:InterruptTeleport()
+            return
+        end
+        if self:GetPower() then
+            if self:GetData("flight") and TARDIS:GetSetting("flight_to_float_if_no_artron", self:GetCreator()) then
+                self:SetData("floatfirst", true)
+            end
+            self:SetPower(false)
+            for k,_ in pairs(self.occupants) do
+                TARDIS:ErrorMessage(k, "Artron Depleted.")
+            end
+        end
+    end)
+
+    local function ArtronDematCheck(self)
+        local fast = self:GetData("demat-fast", false)
+        local artron = self:GetData("artron-val", 0)
+
+        if self:CallHook("ShouldUpdateArtron") == false then return end
+
+        if fast and artron < -artron_values.cost_full then
+            return true
+        end
+        if not fast and artron < -artron_values.cost_demat then
+            return true
+        end
+    end
+
+    ENT:AddHook("ShouldFailDemat", "artron", function(self, force)
+        if not TARDIS:GetSetting("artron_energy") then return end
+        if self:GetData("hads-attempt") then return end
+
+        if ArtronDematCheck(self) == true then
```

Comment:
No need to check == true here
```suggestion
        if ArtronDematCheck(self) then
```

Reply:
Fixed